staging/src/k8s.io/client-go/tools/cache/reflector_test.go

```diff
@@ -1640,6 +1641,95 @@ func getPodListItems(start int, numItems int) (string, string, *v1.PodList) {
 	return out.Items[0].GetName(), out.Items[exemptObjectIndex].GetName(), out
 }
 
+func TestReflectorPanicLogging(t *testing.T) {
```

Comment:
This doesn't really tests logging, to do that we would need to capture logs.

@pohly any recommendations about how to capture logs for testing it's contents?

Reply:
You can use [ktesting.BufferLogs](https://pkg.go.dev/k8s.io/klog/v2/ktesting#BufferLogs) - see example usage in https://github.com/kubernetes/kubernetes/blob/ff25881bc1fe9da1dda4ec82d294b92078ae2763/staging/src/k8s.io/client-go/tools/record/event_test.go#L426-L454